src/modules/map/components/openlayers/OpenLayersMap.vue

```diff
@@ -237,19 +237,46 @@ export default {
         this.map.setView(this.view)
 
         // Click management
-        let pointerDownStart = null
-        let lastClickTimeLength = 0
-        this.map.on('pointerdown', () => {
-            pointerDownStart = performance.now()
-        })
-        // TODO: trigger a click after pointer is down at (roughly) the same spot for longer than 1sec (no need to wait for the user to stop the click)
-        this.map.on('pointerup', () => {
-            lastClickTimeLength = performance.now() - pointerDownStart
-            pointerDownStart = null
-        })
-        // using 'singleclick' event instead of 'click', otherwise a double click (for zooming) on mobile
-        // will trigger two 'click' actions in a row
-        this.map.on('singleclick', (event) => {
+        this.map.on('pointerdown', this.onMapPointerDown)
+        // TODO: trigger a click after pointer is down at (roughly) the same spot
+        // for longer than 1sec (no need to wait for the user to stop the click)
+        this.map.on('pointerup', this.onMapPointerUp)
+        // using 'singleclick' event instead of 'click', otherwise a double click
+        // (for zooming) on mobile will trigger two 'click' actions in a row
+        this.map.on('singleclick', this.onMapSingleClick)
+        this.map.on('pointerdrag', this.onMapPointerDrag)
+        this.map.on('moveend', this.onMapMoveEnd)
+    },
+    beforeUnmount() {
+        this.map.un('pointerdown', this.onMapPointerDown)
+        this.map.un('pointerup', this.onMapPointerUp)
+        this.map.un('singleclick', this.onMapSingleClick)
+        this.map.un('pointerdrag', this.onMapPointerDrag)
+        this.map.un('moveend', this.onMapMoveEnd)
+        this.map.setTarget(null)
+        this.map.setView(null)
+
+        delete this.map
+        delete this.view
```

Comment:
I'm not sure if this is necessary. I'd say that in theory those properties should go away automatically when the component is destroyed.